woocommerce/woocommerce
# Use syncpack to lock pnpm version till we can fix issues

### Changes proposed in this Pull Request:

At present it may be confusing to try install dependencies with newer versions of `pnpm` and find that it introduces issues with unit tests and makes dramatic changes to the lock file. That's because our pnpm engine's value is too lenient right now and we actually can't support pnpm newer than `9.1.3`.

This resolves that issue by fixing the version to `9.1.3` now when you try a `pnpm i` and your version is not exactly `9.1.3` it will cause an error.



### How to test the changes in this Pull Request:

1. On trunk install pnpm version 9.3.0 (`pnpm i -g pnpm@9.3.0`) 
2. run `pnpm i`
3. It will succeed.
4. Now check out this branch and run `pnpm i`.
5. It should fail with an error like:

```
Your pnpm version is incompatible with "[redacted path]".

Expected version: 9.1.3
Got: 9.3.0

This is happening because the package's manifest has an engines.pnpm field specified.
To fix this issue, install the required pnpm version globally.

To install the latest version of pnpm, run "pnpm i -g pnpm".
To check your pnpm version, run "pnpm -v".
```

6. Now change to compliant version (`pnpm i -g pnpm@9.1.3`)
7. Run `pnpm i`
8. It should succeed



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message
Fix pnpm version to 9.1.3 to avoid dependency installation issues.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
